ajax/dropdownConnectNetworkPort.php

```diff
@@ -48,69 +48,98 @@
+      $name_field = 'glpi_netpoints.name';
+      $joins = [
+         'glpi_networkportethernets'   => [
+            'ON'  => [
+               'glpi_networkportethernets'   => 'id',
+               'glpi_networkports'           => 'id'
+            ]
+         ],
+         'glpi_netpoints'              => [
+            'ON'  => [
+               'glpi_networkportethernets'   => 'netpoints_id',
+               'glpi_netpoints'              => 'id'
+            ]
+         ]
+      ];
    }
 
-   $query = "SELECT DISTINCT `glpi_networkports_networkports`.`id` AS wid,
-                             `glpi_networkports`.`id` AS did,
-                             `$table`.`name` AS cname,
-                             `glpi_networkports`.`name` AS nname,
-                             $npname_field AS npname
-             FROM `$table`
-             LEFT JOIN `glpi_networkports`
-               ON (`glpi_networkports`.`items_id` = '".$_POST['item']."'
-                   AND `glpi_networkports`.`itemtype` = '".$_POST["itemtype"]."'
-                   AND `glpi_networkports`.`items_id` = `$table`.`id`
-                   AND `glpi_networkports`.`instantiation_type` = '".$_POST['instantiation_type']."')
-             LEFT JOIN `glpi_networkports_networkports`
-               ON (`glpi_networkports_networkports`.`networkports_id_1` = `glpi_networkports`.`id`
-                   OR `glpi_networkports_networkports`.`networkports_id_2`=`glpi_networkports`.`id`)
-             $JOINS
-             WHERE `glpi_networkports_networkports`.`id` IS NULL
-                   AND `glpi_networkports`.`id` IS NOT NULL
-                   AND `glpi_networkports`.`id` <> '".$_POST['networkports_id']."'
-                   AND `$table`.`is_deleted` = 0
-                   AND `$table`.`is_template` = 0
-             ORDER BY `glpi_networkports`.`id`";
-
-   $result = $DB->query($query);
+   $criteria = [
+      'SELECT'    => [
+         'glpi_networkports_networkports.id AS wid',
+         'glpi_networkports.id AS did',
+         "$table.name AS cname",
+         'glpi_networkports.name AS nname',
+         $name_field
+      ],
+      'DISTINCT'  => true,
+      'FROM'      => $table,
+      'LEFT JOIN' => [
+         'glpi_networkports'  => [
+            'ON'  => [
+               'glpi_networkports'  => 'items_id',
+               $table               => 'id', [
+                  'AND' => [
+                     'glpi_networkports.items_id'           => $_POST['item'],
+                     'glpi_networkports.itemtype'           => $_POST["itemtype"],
+                     'glpi_networkports.instantiation_type' => $_POST['instantiation_type']
+                  ]
+               ]
+            ]
+         ],
+         'glpi_networkports_networkports' => [
+            'ON'  => [
+               'glpi_networkports_networkports' => 'networkports_id_1',
+               'glpi_networkports'              => 'id', [
+                  'OR'  => [
```

Comment:
I'm not sure about this one.
I think it will generate this sql:

```sql
 LEFT JOIN `glpi_networkports_networkports`
ON (`glpi_networkports_networkports`.`networkports_id_1` = `glpi_networkports`.`id`
   AND (`glpi_networkports_networkports`.`networkports_id_2`=`glpi_networkports`.`id`)
)
```

as in `analyseFkey`, for count = 3, the two first key will analyzed recursively and the third passed to analyseCrit

I'm not sure we could achieve an OR condition for a left join actually and i don't have any suggestion so.

Maybe a where part with 
```
'OR' => [
   'glpi_networkports.items_id' => null
   `glpi_networkports_networkports`.`networkports_id_2` => `glpi_networkports`.`id`
]
```

Reply:
```php
$it = $iterator->execute([
   'FROM'      => 'foo',
   'LEFT JOIN' => [
      'bar' => [
         'FKEY'  => [
            'bar' => 'id',
            'foo'    => 'items_id', [
               'OR' => ['field' => 'value']
            ]
         ]
      ]
   ]
]);

echo $it->getSql() . "\n";
//SELECT * FROM `foo` LEFT JOIN `bar` ON (`bar`.`id` = `foo`.`items_id` OR `field` = ?)
```

sounds correct ;)
Also, I'm almost sure we already have existing OR condition in joins.